ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -964,113 +857,8 @@ static bool ShouldHandleFileOpEvent(
-    const char* vnodePath,
-    int pid,
-    const char* procname,
-    int* kauthResult,
-    int* kauthError)
-{
-    // To be useful, the message needs to either provide an FSID/inode pair or a path
-    assert(vnodePath != nullptr || (vnodeFsidInode.fsid.val[0] != 0 || vnodeFsidInode.fsid.val[1] != 0));
-    bool result = false;
-    const char* relativePath = nullptr;
-    
-    OutstandingMessage message =
-    {
-        .receivedResult = false,
-        .rootHandle = root,
-    };
-    
-    if (nullptr != vnodePath)
-    {
-        relativePath = VirtualizationRoot_GetRootRelativePath(root, vnodePath);
-    }
-    
-    int nextMessageId = OSIncrementAtomic(&s_nextMessageId);
-    
-    Message messageSpec = {};
-    Message_Init(
-        &messageSpec,
-        &(message.request),
-        nextMessageId,
-        messageType,
-        vnodeFsidInode,
-        pid,
-        procname,
-        relativePath);
-
-    if (s_isShuttingDown)
-    {
-        return false;
-    }
-    
-    Mutex_Acquire(s_outstandingMessagesMutex);
-    {
-        LIST_INSERT_HEAD(&s_outstandingMessages, &message, _list_privates);
-    }
-    Mutex_Release(s_outstandingMessagesMutex);
-    
-    errno_t sendError = ActiveProvider_SendMessage(root, messageSpec);
-   
-    Mutex_Acquire(s_outstandingMessagesMutex);
-    {
-        if (0 != sendError)
-        {
-            // TODO: appropriately handle unresponsive providers
-            *kauthResult = KAUTH_RESULT_DEFER;
-        }
-        else
-        {
-            while (!message.receivedResult &&
-                   !s_isShuttingDown)
-            {
-                Sleep(5, &message, &s_outstandingMessagesMutex);
-            }
-        
-            if (s_isShuttingDown)
-            {
-                *kauthResult = KAUTH_RESULT_DENY;
-            }
-            else if (MessageType_Response_Success == message.result)
-            {
-                *kauthResult = KAUTH_RESULT_DEFER;
-                result = true;
-            }
-            else
-            {
-                // Default error code is EACCES. See errno.h for more codes.
-                *kauthError = EAGAIN;
-                *kauthResult = KAUTH_RESULT_DENY;
-            }
-        }
-        
-        LIST_REMOVE(&message, _list_privates);
-    }
-    Mutex_Release(s_outstandingMessagesMutex);
-    
-    return result;
-}
-
-static void AbortAllOutstandingEvents()
+static void WaitForListenerCompletion()
 {
-    // Wake up all sleeping threads so they can see that that we're shutting down and return an error
-    Mutex_Acquire(s_outstandingMessagesMutex);
-    {
-        s_isShuttingDown = true;
-        
-        OutstandingMessage* outstandingMessage;
-        LIST_FOREACH(outstandingMessage, &s_outstandingMessages, _list_privates)
-        {
-            wakeup(outstandingMessage);
-        }
-    }
-    Mutex_Release(s_outstandingMessagesMutex);
-    
     // ... and wait until all kauth events have noticed and returned.
```

Comment:
Minor: The comment starting with `// ... and wait` doesn't make as much sense as before.  Maybe updated it to simply "Wait until all kauth events have noticed and returned."

Reply:
Yep, you're right, that doesn't make much sense. Now fixed as suggested.